src/pkg/util/nvidiautils/nvidiautils.go

```diff
@@ -0,0 +1,134 @@
+// Copyright (c) 2018, Sylabs Inc. All rights reserved.
+// This software is licensed under a 3-clause BSD license. Please consult the
+// LICENSE.md file distributed with the sources of this project regarding your
+// rights to use or distribute this software.
+
+package nvidiautils
+
+import (
+	"bufio"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"regexp"
+	"strings"
+)
+
+// GetNvidiaBindPath returns a string array consisting of filepaths of nvidia
+// related files to be added to the BindPaths
+func GetNvidiaBindPath(abspath string) []string {
+	var strArray []string
+	var bindArray []string
+	var searchArray []string
+	var commentID = regexp.MustCompile(`#`)
+	var soID = regexp.MustCompile(".so")
+
+	// use nvidia-container-cli (if present)
+	command, err := exec.LookPath("nvidia-container-cli")
+	if err == nil {
+		cmd := exec.Command(command, "list", "--binaries", "--ipcs", "--libraries")
+		out, err := cmd.Output()
+		if err == nil {
+
+			for _, line := range strings.Split(string(out), "\n") {
+				val := soID.FindString(line) // this will disallow binaries (non .so files)
+				if val != "" {               // contains .so
+					if line != "" {
+						// extract the filename from the path
+						fileNames := strings.SplitAfter(line, "/")
+						fileName := fileNames[len(fileNames)-1]
+
+						testString := line + ":/.singularity.d/libs/" + fileName
+						bindArray = append(bindArray, testString)
+						strArray = append(strArray, fileName)
+					}
+				} else { // binary executable
+					bindArray = append(bindArray, line)
+					strArray = append(strArray, line)
+				}
+			}
+		}
+	}
+
+	cliEntries := strings.Join(strArray, " ") // save away for later comparison check (disallow duplicates)
```

Comment:
Is it necessary to have the second variable `strArray` here?  Can `cliEntries` be constructed from `bindArray`?

Reply:
I don't think so. Because the bind array has entries like /dir/dir/file:/.singularity.d/file in it. I don't think it would be any faster and it would be more work. Are you thinking of a specific method?